metric-anomaly-data-model/src/main/proto/org/hypertrace/alert/engine/eventcondition/config/service/v1/eval_rule.proto

```diff
@@ -2,17 +2,23 @@ syntax = "proto3";
 
 option java_multiple_files = true;
 
-package org.hypertrace.alert.engine.metric.anomlay.rule.config.service.v1;
+package org.hypertrace.alert.engine.eventcondition.config.service.v1;
 
-import "org/hypertrace/metric/anomaly/rule/config/service/v1/alert_rule_enum.proto";
+import "org/hypertrace/alert/engine/eventcondition/config/service/v1/eval_rule_enums.proto";
 
-message MetricAnomalyAlertRule {
+
+message MetricAnomalyEventCondition {
```

Comment:
So I haven't seen any of these, but lots of questions here, including on some of the merged stuff:

- Why does a condition have a name, or labels? Particularly labels seems like it's not a feature that's needed to be built in an MVP.
- Why are the enums in a different file? Can we separate by role rather than message type?
- make sure to snake_case (I see some camel cased merged fields)
- Can we make the period units an enum - or better yet, just a duration iso string?
- Do we support the same types on the LHS and RHS of a filter? Not sure, but it might be worth modeling them separately for flexibility.
- Attribute only has one key - isn't that a key/scope pair (also, can be used in metric selection)
- In the spirit of being iterative, I'd encourage dropping dynamic thresholds for MVP (skipping a lot of questions on them)
- What is an evaluation window? only making a rule enabled at certain times of day? Again - not really an MVP feature, but would also say that belongs at the rule rather than the condition level since it would apply equally across any rule
- not sure what the longevity condition is supposed to do, but I'd guess it belongs in the same clause as the thresholds (see other comment about severity - same idea)
- There's a couple of old issues that look like they may have been copied from one of the original repos where we got stuck with them due to compatibility like:
   - child filter as another field on a filter object (rather than using one of to have leaf filters or composite filters)
   - value having an explicit value type (use one of)

Reply:
> There's a couple of old issues that look like they may have been copied from one of the original repos where we got stuck with them due to compatibility like:
child filter as another field on a filter object (rather than using one of to have leaf filters or composite filters)
value having an explicit value type (use one of)

hey, what issues was the existing model causing for the examples you pointed?

Reply:
> value having an explicit value type (use one of)

So this show up in various bugs - for example, code that was setting value and not setting value type, which other code then switched on. It also means all data types must be serialized to/from a string on either side

> child filter as another field on a filter object (rather than using one of to have leaf filters or composite filters)

This one, I can't remember if we had any actual resulting bugs, but it definitely resulted in confusing logic. The same object could contain a condition as well as children, which doesn't make sense, so you're stuck basically checking if a filter has children and if so, ignoring any conditions (or vice versa) - basically, an undeclared contract to represent something that could be modeled in the api itself.

Reply:
> Why does a condition have a name, or labels? Particularly labels seems like it's not a feature that's needed to be built in an MVP.

Previously, we were considering it as an alert rule proto (independent of notification event proto), so it had a name, and labels. But, after our thought of combining it, yes, this goes at the rule level (NotificationRule).
Labels are metadata, requested by Razorpay during discussion. This is not actionable information, it was meant to add to the notification message (slack). we can add them later, removed from condition proto.

> Why are the enums in a different file? Can we separate by role rather than message type?
make sure to snake_case (I see some camel cased merged fields)

sure, made changes.

> Can we make the period units an enum - or better yet, just a duration iso string?

enum in some cases handy for certain places we want to support (min, sec), but other places (hr, min, sec). But, that implies we need different types of the enum, and changing them will be tricky. So, chose to go with iso string representation.

> Do we support the same types on the LHS and RHS of a filter? Not sure, but it might be worth modeling them separately for flexibility.

Make sense, have made changes.

> Attribute only has one key - isn't that a key/scope pair (also, can be used in metric selection)

On the filter expression, all columns (or attributes) should be from the same scope. And, that can be derived from metric_selection's scope. So, renamed it as ColumnIdentifier.

> In the spirit of being iterative, I'd encourage dropping dynamic thresholds for MVP (skipping a lot of questions on them)

Yup, let's do this. we can bring baseline calculation-based conditions as a follow-up. So, for MVP we can support rules like - `alert me if the duration of service is greater than 30ms`

> What is an evaluation window? only making a rule enabled at certain times of day? Again - not really an MVP feature, but would also say that belongs at the rule rather than the condition level since it would apply equally across any rule

Yes, we can also add this as a follow-up. This also came up during a conversation and it's a requested feature. The use case behind, some customer has different workload during a day, and alert is important for them during certain business hours (say 9am - 9pm) as the same rule might be quite a noise during off hours. 
I was also thinking of doing it two ways,
- Only send alert notifications during this window however continue to evaluate an alert rule
- evaluate Alert Rule only during this time window

> not sure what the longevity condition is supposed to do, but I'd guess it belongs in the same clause as the thresholds (see other comment about severity - same idea)

yes, move that. longevity condition is to express the continuity of a situation for a certain period of time before marking it as an anomalous situation. It helps to express - `alert me if the duration is greater than 30ms for the last 5 mins`

Reply:
Thanks for the changes - the requested use cases make sense, but as long as we make sure we're leaving room for them I think they break off for future iterations cleanly.

Reply:
I really appreciate your help.